packages/api-i18n/src/graphql/context.ts

```diff
@@ -155,31 +168,58 @@ const createBaseContextPlugin = () => {
                 currentLocale = getDefaultLocale();
             }
 
-            __i18n.locale[localeContext] = currentLocale;
+            __i18n.locale.set(localeContext, currentLocale);
 
-            return __i18n.locale[localeContext];
+            return __i18n.locale.get(localeContext);
         };
+
         const getLocales = (): I18NLocale[] => {
             return __i18n.locales;
         };
-        const getLocale = (code: string): I18NLocale | null => {
+
+        const getLocale = (code: string) => {
             const item = __i18n.locales.find(
                 locale => locale.code.toLowerCase() === code.toLowerCase()
             );
+
             if (!item) {
-                return null;
+                return undefined;
             }
             return item;
         };
 
+        const setCurrentLocale: I18NContextObject["setCurrentLocale"] = (localeContext, locale) => {
+            __i18n.locale.set(localeContext, locale);
+        };
+
+        const checkI18NContentPermission = async () => {
+            if (await hasI18NContentPermission(context)) {
+                return;
+            }
+
+            throw new NotAuthorizedError();
+        };
+
+        const setContentLocale: I18NContextObject["setContentLocale"] = locale => {
+            setCurrentLocale("content", locale);
+        };
+
+        const getContentLocale: I18NContextObject["getContentLocale"] = () => {
+            return getCurrentLocale("content");
+        };
```

Comment:
```suggestion
        const getContentLocale: I18NContextObject["getContentLocale"] = getCurrentLocale("content");
```

Reply:
Your suggestion is not really the same. What I'm doing in the code is creating a new arrow function, which will return the current content locale when invoked. Your suggestion invokes it immediately, which defeats the purpose of it. We need the ability to get the content locale at any point in time, and that can also be after the current locale is changed. So in short - we need a function, not a value.

Reply:
Ah yes my bad, I went a bit too far with my suggestion it seems.